### ISSUE TITLE
Sofort and Giropay gateways not allowing checkout

### ISSUE THREAD

**Issue by frosso** [Automattic/woocommerce-payments#2185]
## Description

It looks like the Sofort and Giropay gateways (NOT UPE) are no longer working.
User is blocked by an error at checkout stating: "Invalid payment method. Please input a new card number."
This appears to be because the `Payment_Information` class is unable to pull the correct payment method from the `POST` request (perhaps the correct key needs to be added to [this list](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-payment-information.php#L220), but we haven't investigated this one too deeply).

Investigate why.

**Comment by naman03malhotra** [comment]
It doesn't look relevant anymore, but old PMs can use a code cleanup.
cc @jessy-p

**Comment by jessy-p** [comment]
@marcinbot  has done the cleanup in https://github.com/Automattic/woocommerce-payments/pull/3394 and the issue [Remove redundant payment gateways](https://github.com/Automattic/woocommerce-payments/issues/2659) has been closed. cc: @vbelolapotkov , @naman03malhotra

**Comment by vbelolapotkov** [comment]
I definitely see a bunch of mentions for SEPA, Giropay & Sofort here: https://github.com/Automattic/woocommerce-payments/blob/8926d370a05c49382b3149300b0102f7a7d3ca75/client/checkout/classic/index.js#L190-L190

@jessy-p I guess those need to be cleaned up too?

**Comment by jessy-p** [comment]
Thanks @vbelolapotkov  Created issue https://github.com/Automattic/woocommerce-payments/issues/4519 for further code cleanup, particularly JS code.
